scp/cc/cpio/client_providers/kms_client_provider/src/azure/azure_kms_client_provider.cc

```diff
@@ -162,10 +255,36 @@ void AzureKmsClientProvider::GetSessionCredentialsCallbackToDecrypt(
       hasSnp() ? fetchSnpAttestation() : fetchFakeSnpAttestation();
   CHECK(report.has_value()) << "Failed to get attestation report";
 
+  EVP_PKEY* publicKey;
+  EVP_PKEY* privateKey;
+  if (hasSnp()) {
+    std::cout << "Generate wrapping Key on SNP" << std::endl;
+    // Generate wrapping key
+    wrappingKey_ = AzurePrivateKeyFetchingClientUtils::GenerateWrappingKey();
+    privateKey = wrappingKey_.first;
+    publicKey = wrappingKey_.second;
+  } else {
+    // Get test PEM public key
+    auto publicPemKey = GetTestPemPublicWrapKey();
+    publicKey = AzurePrivateKeyFetchingClientUtils::PemToEvpPkey(publicPemKey);
+
+    // Get test PEM private key and convert it to EVP_PKEY*
+    auto privateKeyPem = GetTestPemPrivWrapKey();
+    privateKey = nullptr;
+    BIO* bio = BIO_new_mem_buf(privateKeyPem.c_str(), -1);
+    PEM_read_bio_PrivateKey(bio, &privateKey, nullptr, nullptr);
```

Comment:
I know the private key is hard coded, but having a check for the result of PEM_read_bio_PrivateKey would make our life easier when a problem happens. I think you can use `CHECK` which is used for assertion in the code base.